samples/basic/rgb_gradient/README.rst

```diff
@@ -0,0 +1,56 @@
+.. zephyr:code-sample:: rgb-gradient
+
+RGB PWM Gradient
+################
```

Comment:
the directive calls for more actual metadata, and the name of the sample is redundant with the heading

```suggestion
.. zephyr:code-sample:: rgb-gradient
   :name: PWM RGB Gradient
   :relevant-api: pwm_interface

   Drive an RGB LED an RGB LED in a color gradient pattern using PWM.
```